lib/auth/saml.go

```diff
@@ -44,46 +44,68 @@ import (
 	"github.com/gravitational/teleport/lib/utils"
 )
 
+// ErrSAMLNotImplemented is the error returned by the SAML methods when not
+// using the Enterprise edition of Teleport.
+var ErrSAMLNotImplemented = trace.AccessDenied("SAML is only available in enterprise subscriptions")
```

Comment:
Good catch. That's because I factored the error to the top-level, there's no (useful) trace information, so I should now wrap at the point the error is returned? Otherwise if I had `return trace.AccessDenied(errMsg)`, that would wrap correctly?

Also, were you also asking me to rename the var? I'll rename to `ErrSAMLRequiresEnterprise` since the error message contains `SAML` (and will conflict with OIDC when I do that next).

Reply:
No, not asking you to rename.

Basically I want `errors.Is(err, ErrRequiresEnterprise)` to return true for whatever `err` it is you return. This way we have a consistent way of checking whether an error is due to requiring enterprise without resorting to string comparisons.

Reply:
ack. sorry I went looking for `ErrRequiresEnterprise` but didn't find it - hence my confusion - I must have mistyped. Will update.

Reply:
done